### PR TITLE
More accurate mask filling. improve mask acc by 0.2-0.3 mAP

### PR DESCRIPTION
**Pull request by Jerryzcn**
add more accurate mask filling. improve mask acc by 0.2-0.3 mAP
adopted from detectron2
this is only used in inference.

### CODE REVIEW

File: gluoncv/data/transforms/mask.py [hunk 1]
```diff
@@ -125,22 +131,31 @@ def fill(mask, bbox, size):
...
+    hh = hh * (float(M + 2) / M)
     x1, y1, x2, y2 = x - hw, y - hh, x + hw, y + hh
-    # quantize
-    x1, y1, x2, y2 = map(int, (x1, y1, x2, y2))
-    w, h = (x2 - x1 + 1), (y2 - y1 + 1)
+    if fast_fill:
+        # quantize
+        x1, y1, x2, y2 = map(int, (x1 + 0.5, y1 + 0.5, x2 + 0.5, y2 + 0.5))
+        w, h = (x2 - x1 + 1), (y2 - y1 + 1)
+        mask = mx.nd.array(mask)
```
Comment (zhreshold) [inline gluoncv/data/transforms/mask.py]: change to use PIL resize directly?

**Comment by mli** [comment]
Job PR-997-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-997/1/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/data/transforms/mask.py (hunk 1 above)
Comment (Jerryzcn) [reply gluoncv/data/transforms/mask.py]: is there any issue with mx imresize?

Comment (zhreshold) [reply gluoncv/data/transforms/mask.py]: nothing bad with imresize, but numpy.ndarray->NDArray->numpy.ndarray requires two memcpy

**Comment by mli** [comment]
Job PR-997-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-997/2/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**